src/schema/v2/filterArtworksConnection.ts

```diff
@@ -533,12 +538,22 @@ const filterArtworksConnectionTypeFactory = (
       include_artworks_by_followed_artists ||
       requestedPersonalizedAggregation
     ) {
-      if (!loaderWithoutCache) {
+      if (!filterArtworksAuthenticatedLoader) {
         throw new Error("You must be logged in to request these params.")
       }
-      loader = loaderWithoutCache
+      loader = filterArtworksAuthenticatedLoader
     } else {
-      loader = loaderWithCache
+      // If filtering by sale and sorting by price,
+      // use the uncached loader to avoid stale data.
+      const isSortingByPrice = ["-prices", "prices"].includes(
+        gravityOptions.sort
+      )
+      const isFilteringBySale = !!gravityOptions.sale_id
+      if (isSortingByPrice && isFilteringBySale) {
```

Comment:
Another thought, which we can discuss but feels 'correct' to do (can be added later so not a blocker) - is to augment this to be `if (isFilteringBySale && (isSortingByPrice || isFilteringByPrice))` - allow filtering by price to skip the cache as well. That way, when using the price filter, the results should be consistent with the up-to-date information displayed in the artwork brick's tombstone info (very analogous to the bug report about sorting).

That can be a follow-up if we choose to do it.

Reply:
I won't add that yet, but once this looks to be working better re: sorting, will add `isFilteringByPrice`